### PR TITLE
cleanup: prevent behavior change from a gRPC change

### PR DESCRIPTION
**Pull request by jacobsa**
This class currently encourages an ownership hack that works around a gRPC API
issue that is [being fixed](https://github.com/grpc/proposal/pull/238). At that
point the tests that use this class will start failing, because they manage it
using **two different `std::unique_ptr` objects**.

Prevent them failing by overriding the upcoming `Destroy` method to do nothing.
This preserves the existing behavior. Because I haven't used the `override`
keyword, this works before and after the gRPC change, decoupling this issue
from the import of the gRPC change.

Longer term it would be better to make the use of this class not require the
ownership hack at all, but that is a larger project. In the meantime this
change unblocks fixing the need for such hacks for other subclasses that don't
want to further promulgate the issue.

@coryan @devbww

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6565)

### CODE REVIEW

File: google/cloud/bigtable/testing/mock_response_reader.h [hunk 1]
```diff
@@ -111,6 +113,15 @@ class MockAsyncResponseReader
   MOCK_METHOD(void, StartCall, (), (override));
   MOCK_METHOD(void, ReadInitialMetadata, (void*), (override));
   MOCK_METHOD(void, Finish, (Response*, grpc::Status*, void*), (override));
+
+  // Preserve the behavior from before https://github.com/grpc/proposal/pull/238
+  // of not destroying the object when
+  // std::unique_ptr<grpc::ClientAsyncResponseReaderInterface> goes out of
+  // scope.
+  //
+  // TODO(jacobsa): mark this with the override keyword once the method exists
```
Comment (coryan) [inline google/cloud/bigtable/testing/mock_response_reader.h:122]: nit: we do not create `TODO(person)`, I created #6566 for your, please change this to `// TODO(#6566)`

**Review by coryan — COMMENTED** [review]
LGTM, waiting for the builds to approve and merge.

**Comment by coryan** [comment]
The publicly accessible build logs will be up in a few minutes, but the gist of the `checkers-pr` failure is:

```
/workspace/google/cloud/bigtable/testing/mock_response_reader.h:122:11 - Unknown word (jacobsa)
```

Which I already asked you to change.

File: google/cloud/bigtable/testing/mock_response_reader.h (hunk 1 above)
Comment (jacobsa) [reply google/cloud/bigtable/testing/mock_response_reader.h]: I was wondering about that. Done, thanks.

**Comment by jacobsa** [comment]
@coryan: Fixed the TODO. Do I need to do something to make the checks run again? Sorry, unfamiliar with the process.

**Comment by jacobsa** [comment]
@coryan: sorry, I think you need to do it once more after I fixed the commit message lint error.

**Comment by coryan** [comment]
Unfortunately you look like an external contributor from Google Cloud Build's point of view, one of us needs to trigger them.

the conventionalcommits.org check is annoying (but we can ignore it), it is complaining about your **commit** title.  I have found that a second commit makes it not care and just use the PR title (as it should). Leave it be, I can merge with the failure outstanding.

**Comment by coryan** [comment]
Ooops, I forgot about kicking off the Kokoro builds too.

**Review by coryan — APPROVED** [review]

**Comment by coryan** [comment]
Thanks for the PR!

**Comment by jacobsa** [comment]
Thanks!

**Pull request merged**